### ISSUE TITLE
The 'GLOBAL IN' clause doesn't work as expected with a subquery

### ISSUE THREAD

**Issue by rbushri** [ClickHouse/ClickHouse#56790]
I created a simple distributed table (3 shards) , inserted 100 rows and tried to execute the following query. 
I expected to get 100 rows, but the results contained fewer than 100 rows. It seems ClickHouse retrieved data only from 2 shards instead of all the shards.

(My original example is more complicated, so I've attached a simplified version of the problem below.)

**How to reproduce:**

ClickHouse version: 23.8.3.48

Create a distributed table and insert 100 rows:
```
CREATE TABLE test_local ON CLUSTER '{cluster}' (
    x Int64,
)
ENGINE = ReplicatedMergeTree;


CREATE TABLE test ON CLUSTER '{cluster}' AS test_local
ENGINE = Distributed('{cluster}', default, test_local, rand());


insert into test select * from numbers(100);
```
select the data from the table :
```
select * from test
where 'XXX' global in (select 'XXX');
```
Expected: 100 rows, but returned fewer.

**Expected behavior:** Return 100 rows.

**Comment by den-crane** [comment]
Can you please share 

```
select name,value from system.settings where changed
```

**Comment by rbushri** [comment]
connect_timeout_with_failover_ms	1000
load_balancing	nearest_hostname
distributed_aggregation_memory_efficient	1
do_not_merge_across_partitions_select_final	1
os_thread_priority	2
log_queries	1
max_result_rows	20000
result_overflow_mode	break
parallel_view_processing	1
flatten_nested	0
format_avro_schema_registry_url	http://xxx.com:8081

**Comment by den-crane** [comment]
max_result_rows  / result_overflow_mode break

this settings makes SELECT to stop silently. Try without them. Or try `select count() from test where 'XXX' global in (select 'XXX');`

**Comment by den-crane** [comment]
Right the issue exists 23.8.3.48 https://fiddle.clickhouse.com/a629332a-f8c3-43b6-9ea6-ba2357f4877a
But not exists in 23.8.6 https://fiddle.clickhouse.com/af98613c-2392-4237-839b-c72fc3090d49
You should upgrade to 23.8.6.16

**Comment by den-crane** [comment]
the issue started in 23.6, fixed in 23.8.4
probably related to PREWHERE https://github.com/ClickHouse/ClickHouse/issues/51846

**Comment by rbushri** [comment]
Thanks a lot! I upgraded the version and now it's working.
